### ISSUE TITLE
--show-all is not showing all frames

### ISSUE THREAD

**Issue by holmanb** [joerick/pyinstrument#130]
```
$ ~/.local/bin/pyinstrument --load-prev 2021-05-01T11-59-28 --show-all

  _     ._   __/__   _ _  _  _ _/_   Recorded: 11:59:28  Samples:  44565
 /_//_/// /_\ / //_// / //_'/ //     Duration: 47.341    CPU time: 46.270
/   _/                      v3.4.1

Program: /usr/lib/python-exec/python3.8/emerge -pv @world

47.341 <module>  ../../bin/emerge:11
└─ 47.332 <module>  ../../workspace/portage/<string>:5
   └─ 46.930 emerge_main  _emerge/main.py:1210
         [11040 frames hidden]  _emerge, portage, copy, copyreg, posi...

To view this report with different options, run:
    pyinstrument --load-prev 2021-05-01T11-59-28 [options]

```

similarly, `--show <pattern>` is not showing the frames I'm interested in.

**Comment by joerick** [comment]
Hey! I just pushed a fix in a PR #131 - would you mind giving it a shot? You should be able to install it by doing

    pip install https://github.com/joerick/pyinstrument/archive/refs/heads/fix-130.zip

**Comment by holmanb** [comment]
Thanks for the quick response! I just tried that but don't see any difference:
```
$ python -m pyinstrument --show-all /usr/bin/emerge -pv @world
...
  _     ._   __/__   _ _  _  _ _/_   Recorded: 02:55:37 PM Samples:  54086
 /_//_/// /_\ / //_// / //_'/ //     Duration: 57.096    CPU time: 56.936
/   _/                      v3.4.1

Program: /usr/lib/python-exec/python3.8/emerge -pv @world

57.096 <module>  emerge:11
└─ 57.089 <module>  ../../<string>:5
   └─ 56.563 emerge_main  _emerge/main.py:1210
         [11914 frames hidden]  _emerge, portage, copy, copyreg, posi...

To view this report with different options, run:
    pyinstrument --load-prev 2021-05-01T14-55-37 [options]

$ ~/workspace/portage $ ~/.local/bin/pyinstrument --version
pyinstrument 3.4.1, on Python 3.8.8
```

**Comment by holmanb** [comment]
Actually it looks like the version currently installed (via the commands above) doesn't match your pull request, so I wasn't testing the new version.  I'll try to figure that out, stay tuned.

**Comment by holmanb** [comment]
After reinstalling, it seems to be working as expected. Thanks!

**Comment by joerick** [comment]
Released as v3.4.2
